arangod/Aql/IResearchViewNode.cpp

```diff
@@ -464,27 +504,9 @@ bool hasDependencies(aql::ExecutionPlan const& plan, aql::AstNode const& node,
       // unable to find setter
       continue;
     }
-
-    if (!setter->isDeterministic()) {
-      // found nondeterministic setter
+    if (isSetterVolatile(plan, *setter)) {
```

Comment:
If we'll decide to stick with local helper function instead of `getLoop` I'd prefer to pass a container for setter variables instead of instantiating it for every call.